File: myBackend/app/models/userModel.py
from pydantic import BaseModel, EmailStr, Field
from typing import Annotated, Optional
from datetime import date


class UserRegister(BaseModel):
    email: EmailStr
    password: Annotated[str, Field(min_length=8)]
    first_name: str
    last_name: str
    dob: date
    phone_number: Optional[Annotated[str, Field(min_length=11, max_length=11)]]


class UserLogin(BaseModel):
    email: EmailStr
    password: Annotated[str, Field(min_length=8)]


class UserOut(BaseModel):
    user_id: int
    first_name: str
    last_name: str
    email: EmailStr

File: my_frontend/lib/models/user_model.dart
class UserProfile {
  final int userId;
  final String firstName;
  final String lastName;
  final String email;

  UserProfile({
    required this.userId,
    required this.firstName,
    required this.lastName,
    required this.email,
  });

  factory UserProfile.fromJson(Map<String, dynamic> json) => UserProfile(
    userId: json['user_id'],
    firstName: json['first_name'],
    lastName: json['last_name'],
    email: json['email'],
  );
}